include/sycl/algorithm/reduce.hpp

```diff
@@ -94,6 +100,343 @@ typename std::iterator_traits<Iterator>::value_type reduce(
                                      cl::sycl::access::target::host_buffer>();
   return bop(hI[0], init);
 }
+#else
+size_t up_rounded_division(size_t x, size_t y){
+  return (x+(y-1)) / y;
+}
```

Comment:
This function is defined on the reduce.hpp header, but used on the inclusive_scan.hpp header. This creates a dependency among the two which I am not convinced it is really necessary. If you have common stuff across two algorithms, you can put them on the algorithm_composite_patterns, or you can create your own header with commonalities across triSYCL variants of the implementations.